cmd/update/volume_move.go

```diff
@@ -0,0 +1,378 @@
+			for _, rep := range vol.Replicas {
+				if rep.Node.String() == c.toNode {
+					c.display.TextMessage(ctx, c.writer, fmt.Sprintf("new replica added on node %s, starting sync", c.toNode))
+					return nil
+				}
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) waitForReplicaOnNodeToSync(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			// find the new deployment added with this operation
+			for _, rep := range vol.Replicas {
+				if rep.Node.String() == c.toNode {
+					if rep.Health == health.ReplicaReady {
+						return nil
+					} else {
+						c.display.ReplicaSyncProgress(ctx, c.writer, *rep.SyncProgress)
+					}
+				}
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) waitForMasterPromote(ctx context.Context, nsID id.Namespace) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timed out while watching for desired state but the operation may still be running (server side)")
+
+		case <-time.After(time.Second * 5):
+			vol, err := c.getVolume(ctx, nsID)
+			if err != nil {
+				return fmt.Errorf("failed to get updated volume config from api: %s", err)
+			}
+
+			if vol.Master.Node.String() == c.toNode && vol.Master.Health == health.MasterOnline {
+				return nil
+			}
+		}
+	}
+}
+
+func (c *volumeMoveCommand) getVolume(ctx context.Context, nsID id.Namespace) (vol *model.Volume, err error) {
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return nil, err
+	}
+
+	if useIDs {
+		vol, err = c.client.GetVolume(ctx, nsID, id.Volume(c.volumeID))
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	} else {
+		vol, err = c.client.GetVolumeByName(ctx, nsID, c.volumeID)
+		if err != nil {
+			return nil, err
+		}
+		return vol, nil
+	}
+}
+
+func newMoveCmd(w io.Writer, client interfaces.Client, config interfaces.ConfigProvider) *cobra.Command {
+	c := &volumeMoveCommand{
+		config: config,
+		client: client,
+		writer: w,
+	}
+
+	cobraCommand := &cobra.Command{
+		Use:   "move [volume name] [source node] [destination node]",
+		Short: "Updates a volume's placement",
+		Example: `
++$ storageos update volume move my-volume-name source-node-id destination-node-id --namespace my-namespace-name
++$ storageos update volume move pvc-6d531e0e-fc5a-4b35-b752-bf75fc5dc690 38b7f402-c602-49f9-96b7-bc213d02a737 d4e910ac-394b-4fdb-be51-27822c15b682
+`,
+		Args: argwrappers.WrapInvalidArgsError(func(cmd *cobra.Command, args []string) error {
+			if len(args) != 3 {
+				return clierr.NewErrInvalidArgNum(args, 3, "storageos update volume move [volume] [source node] [destination node]")
+			}
+
+			c.volumeID = args[0]
+			c.fromNode = args[1]
+			c.toNode = args[2]
```

Comment:
Looks like we're still caring about both arguments and flags, I think we should use flags

Reply:
We will use a mix of the two just like on the other commands of the cli

An example of this is `storageos get volume --namespace my-namespace-name my-volume-name`

We will make use the global flags like `namespace` and `timeout` and the cmd itself will use and require 3 args

So a typical call would look like `storageos update volume move vol-id-123123123121 from-node-id-123123112 destination-node-id-123123123 --namespace my-namespace-id-123123 --timeout 10m`